addons/account_edi_proxy_client/models/account_edi_proxy_user.py

```diff
@@ -64,10 +65,25 @@ class AccountEdiProxyClientUser(models.Model):
 
     _sql_constraints = [
         ('unique_id_client', 'unique(id_client)', 'This id_client is already used on another user.'),
-        ('unique_edi_identification', 'unique(edi_identification, proxy_type, edi_mode)', 'This edi identification is already assigned to a user'),
-        ('unique_company_proxy', 'unique(company_id, proxy_type, edi_mode)', 'This company has a user already created for this EDI type'),
+        ('unique_active_edi_identification', '', 'This edi identification is already assigned to an active user'),
```

Comment:
Does this work?
```suggestion
        ('unique_active_edi_identification',
         'EXCLUDE (company_id WITH =, proxy_type WITH =, edi_mode WITH =) WHERE (active)',
         'This edi identification is already assigned to an active user'),
```
Ref https://www.postgresql.org/docs/15/sql-createtable.html#SQL-CREATETABLE-EXCLUDE

If yes, it may avoid the need for the `_auto_init`.

Reply:
it does work, that's how I wrote it originally in https://github.com/odoo/odoo/pull/129595#discussion_r1275120223 but I think `_auto_init` is a preferred in such situation, so I changed it

Reply:
Interesting. I didn't know about the virtual constraints. Thanks.
It's OK for me.